two_factor/views/core.py

```diff
@@ -213,7 +213,10 @@ class SetupView(IdempotentSessionWizardView):
         ('validation', DeviceValidationForm),
         ('yubikey', YubiKeyDeviceForm),
     )
+
+    show_welcome = not hasattr(settings, 'TWO_FACTOR_SKIP_WELCOME') or not settings.TWO_FACTOR_SKIP_WELCOME
```

Comment:
2. It's only evaluated once which makes testing harder. When you want to check for different values of `TWO_FACTOR_SKIP_WELCOME` you'd need to take care of that yourself. Using dynamic evaluation at runtime isn't much slower within this simple situation but makes testing easier.